src/cpu/aarch64/jit_sve_512_x8s8s32x_conv_kernel.hpp

```diff
@@ -0,0 +1,296 @@
+
+private:
+    const int ic_sub_step = 4;
+
+    enum {
+        typesize = sizeof(float),
+        ker_reg_base_idx = 28,
+        ker_dw_reg_base_idx = 30,
+        ker_zp_reg_base_idx = 26,
+    };
+    typedef enum {
+        no_last_block,
+        last_ic_block,
+        last_sp_block,
+    } ic_block_t;
+
+    /* data regs */
+    const XReg reg_ptr_scales = x7; //rax;
+    const XReg aux_reg_saturation = x7; //rax;
+    const XReg reg_inp = x8; //r8;
+    const XReg reg_ker = x9; //r9;
+    const XReg reg_out = x10; //r10;
+    const XReg aux_reg_inp = x11; //r11;
+    const XReg reg_ptr_sum_scale = x11; //r11;
+    const XReg aux_reg_ker = x12; //r12;
+    const XReg reg_compensation = x14; //r14;
+    const XReg aux_reg_inp_d = x13; //r13;
+    const XReg aux_reg_ker_d = x15; //r15;
+    // Using 3d regs as depthwise_3d is not yet supported
+    const XReg reg_inp_buffer_ptr = aux_reg_inp_d;
+    const XReg aux_reg_inp_buffer_ptr = aux_reg_ker_d;
+    // zero-point computation
+    const XReg reg_zp_compensation = aux_reg_inp;
+    const XReg reg_src_zero_point = aux_reg_ker_d;
+    const XReg reg_dst_zero_point = reg_src_zero_point;
+
+    /* counter regs */
+    const XReg reg_bias_alpha = x1; //abi_not_param1;
+    const XReg reg_param1 = x0; //abi_param1;
+    const XReg reg_oi = x3; //rbx;
+    const XReg reg_bias = x2; //rdx;
+    const XReg reg_oc_blocks = x6; //rsi;
+    const XReg reg_owb = aux_reg_ker;
+    const XReg reg_scratch = reg_compensation;
+    const XReg reg_kj = reg_ptr_scales;
+    const XReg reg_ki = reg_compensation;
+    const XReg reg_overflow = reg_ptr_scales;
+    const XReg reg_icb = reg_bias;
+    const XReg reg_jmp_tbl_base = reg_kj;
+
+    XReg reg_stack = x22; // translator stack register
+
+    /* Temporay registers */
+    XReg reg_tmp0_imm = x18; // tmp for add_imm
+    XReg reg_tmp1_imm = x19; // tmp for add_imm
+    XReg reg_tmp2_imm = x20; // tmp for add_imm
+    XReg reg_tmp3_imm = x21; // tmp for add_imm
+    XReg reg_tmp0_adr = x23; // tmp for address value
+    XReg reg_tmp1_adr = x24; // tmp for address value
+    XReg reg_tmp2_adr = x25; // tmp for address value
+    XReg reg_tmp3_adr = x26; // tmp for address value
+
+    const PReg ktail_mask = p2;
+    const PReg kblend_mask = p8;
+
+    const PReg mask_tmp = p3;
+    const PReg mask_tmp2 = p9;
+    const PReg mask_all_one = p4;
+
+    const ZReg vmm_wei = ZReg(31);
+    /* used during bias section of store_output */
+    const ZReg vmm_comp = ZReg(30); // only for signed input
+    const ZReg vmm_bias = ZReg(31);
+    /* used during post_op sum section of store_output */
+    const ZReg vmm_prev_dst = ZReg(31);
+    /* used during write-out section of store_output */
+    const ZReg vmm_saturation = ZReg(30);
+    const ZReg vmm_zero = ZReg(31);
+
+    /* used in compute_ker (but set during prepare_output) */
+    const ZReg vmm_shift = vmm_comp; // only for signed input
+    /* used in compute_ker (but only for pre-VNNI machines) */
```

Comment:
As I understand unlike `x64` here we don't make any distinction between `vnni` and `pre-vnni` so probably above comment does not belong here

Reply:
Thx! I'll delete it.